stock_storage_type/tests/test_storage_type_move.py

```diff
@@ -38,28 +38,33 @@ def _test_confirmed_move(self, product=None):
         return move_to_assign
 
     def test_not_only_empty_confirmed_move(self):
-        self.pallets_location_storage_type.write({"allow_new_product": "mixed"})
+        category = self.pallets_location_storage_type.storage_category_id
+        category.allow_new_product_ids.allow_new_product = "mixed"
         move = self._test_confirmed_move()
         self.assertEqual(
             move.move_line_ids.location_dest_id, self.pallets_bin_1_location
         )
 
     def test_only_empty_confirmed_move(self):
-        self.pallets_location_storage_type.write({"allow_new_product": "empty"})
+        category = self.pallets_location_storage_type.storage_category_id
+        category.allow_new_product_ids.allow_new_product = "empty"
         move = self._test_confirmed_move()
         self.assertNotEqual(
             move.move_line_ids.location_dest_id, self.pallets_bin_1_location
         )
 
     def test_do_not_mix_products_confirmed_move_ok(self):
-        self.pallets_location_storage_type.write({"allow_new_product": "same"})
+        category = self.pallets_location_storage_type.storage_category_id
+        category.allow_new_product_ids.allow_new_product = "same"
         move = self._test_confirmed_move()
         self.assertEqual(
             move.move_line_ids.location_dest_id, self.pallets_bin_1_location
         )
 
     def test_do_not_mix_products_confirmed_move_nok(self):
-        self.pallets_location_storage_type.write({"allow_new_product": "same"})
+        self.pallets_location_storage_type.storage_category_id.write(
```

Comment:
Was a bit brutal with this change, this doesn't really comply with what we are testing, need to fix that:

- set the "Allow new product" policy through a rule, not globally on the storage category (especially true for tests on cardboxes with different lots below in this file)